### PR TITLE
LPS-196493 

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPS-196493

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Review by Hijky — APPROVED** [review]

**Comment by Hijky** [comment]
FYI, for the APIApplicationEndpoint.macro, we also have a created macro named EndpointAPI.macro, how about merging them in the same macro

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro [hunk 1]
```diff
@@ -32,6 +34,13 @@ definition {
 		Variables.assertDefined(parameterList = "${baseURL},${status},${title}");
 
 		var curl = ApplicationAPI._curlAPIApplication(virtualHost = ${virtualHost});
+
```
Comment (Hijky) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro:37]: Could you take a look at it? I'm not sure where can we use externalReferenceCodeApp, it only appears in this if the function

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro [hunk 2]
```diff
@@ -32,6 +34,13 @@ definition {
...
+		if (!(isSet(externalReferenceCodeApp))) {
```
Comment (Hijky) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro:38]: Could you take a look at it? I'm not sure where can we use externalReferenceCodeApp, it only appears in this if the function

**Comment by magjed4289** [comment]
ci:forward

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro (hunk 2 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro:38]: You may be right we don't use this one in this test, I added it by default together with endpoint erc and schema erc so it's always set up to custom, for sake of future tests :)

**Pull request closed without merging**